crates/rolldown/src/module_loader/module_task.rs

```diff
@@ -137,7 +137,11 @@ impl ModuleTask {
     let repr_name = self.resolved_id.id.as_path().representative_file_name().into_owned();
     let repr_name = legitimize_identifier_name(&repr_name);
 
-    let id = ModuleId::new(ArcStr::clone(&self.resolved_id.id));
+    // `ModuleId` is the filename we used to create a file, replace `\0` to avoid raise issue,
+    // ref: https://en.wikipedia.org/wiki/Filename#Reserved_characters_and_words
+    // TODO: maybe we need to handle all reserved characters, but `\0` is widely used reserved
+    // char
+    let id = ModuleId::new(self.resolved_id.id.replace('\0', "_"));
```

Comment:
I investigate a little bit. The cause is `\0` is invalid for os's file system to use. Fix here https://github.com/rolldown/rolldown/pull/2664.

Reply:
> This is probably not a proper workaround fix. This will cause case like `load(id)( if (id.starts_with('\0virtual')) return "code" )` failed.

I don't think so, if in your case, the test case will fail

Reply:
Yeah. Because this where `resolveId` and `load` hook are already called.

However, it's still changes the the actually content of the id. Many logic rely that behaviors would failed due to this.

For example, people record a virtual id like `\0test` in load hook, and he can't find the module the final chunks`. Because the final chunk contains a module called `_test`  instead of `\0test`.